pipeline.py

```diff
@@ -111,32 +110,65 @@ def pipeline_wrapper(mock: List[Literal[MOCKED_STAGES]]):
         description="InstructLab pipeline",
     )
     def pipeline(
-        num_instructions_to_generate: int = 2,
-        repo_url: str = "https://github.com/instructlab/taxonomy.git",
-        repo_branch: Optional[str] = None,
-        repo_pr: Optional[int] = None,
-        storage_class_name: str = "nfs-csi",
-        base_model: str = BASE_MODEL,
-        # minimal subset of MMLU_TASKS
-        # mmlu_tasks_list: str = MMLU_TASKS_LIST,
-        model_dtype: str = MODEL_DTYPE,
-        few_shots: int = FEW_SHOTS,
-        batch_size: str = BATCH_SIZE,
-        max_workers: str = MAX_WORKERS,
-        merge_system_user_message: bool = MERGE_SYSTEM_USER_MESSAGE,
-        device: str = None,
-        nproc_per_node: int = 3,
-        nnodes: int = 2,
-        num_epochs_phase_1: int = NUM_EPOCHS_PHASE_1,
-        num_epochs_phase_2: int = NUM_EPOCHS_PHASE_2,
-        effective_batch_size: int = EFFECTIVE_BATCH_SIZE,
-        learning_rate: float = LEARNING_RATE,
-        num_warmup_steps: int = NUM_WARMUP_STEPS,
-        save_samples: int = SAVE_SAMPLES,
+        # SDG phase
+        sdg_repo_url: str = "https://github.com/instructlab/taxonomy.git",
+        sdg_repo_branch: Optional[str] = None,
+        sdg_repo_pr: Optional[int] = None,
+        sdg_base_model: str = BASE_MODEL,
+        sdg_scale_factor: int = 2,  # Renamed upstream https://github.com/instructlab/instructlab/blob/f7d40f6ed5112d59132dd832bd332fa6fbbe7010/src/instructlab/configuration.py#L279-L290
         sdg_pipeline: str = SDG_PIPELINE,
-        max_batch_len: int = MAX_BATCH_LEN,
-        seed: int = SEED,
+        sdg_max_batch_len: int = MAX_BATCH_LEN,
+        # Training phase
+        train_nproc_per_node: int = 3,
+        train_nnodes: int = 2,
+        train_num_epochs_phase_1: int = NUM_EPOCHS_PHASE_1,
+        train_num_epochs_phase_2: int = NUM_EPOCHS_PHASE_2,
+        train_effective_batch_size: int = EFFECTIVE_BATCH_SIZE,
+        train_learning_rate: float = LEARNING_RATE,
+        train_num_warmup_steps: int = NUM_WARMUP_STEPS,
+        train_save_samples: int = SAVE_SAMPLES,
+        train_max_batch_len: int = MAX_BATCH_LEN,
+        train_seed: int = SEED,
+        # MT Bench
+        mt_bench_max_workers: str = MAX_WORKERS,
+        # Final evaluation
+        eval_few_shots: int = FEW_SHOTS,
+        eval_batch_size: str = BATCH_SIZE,
+        # Other options
+        merge_system_user_message: bool = MERGE_SYSTEM_USER_MESSAGE,
```

Comment:
This paramater is used for `mt_bench`
```suggestion
        mt_bench_merge_system_user_message: bool = MERGE_SYSTEM_USER_MESSAGE,
```

Reply:
Yes, but it's used for MT Bench Branch in final eval as well:

https://github.com/opendatahub-io/ilab-on-ocp/blob/4ed3565cc053e4a65298c92c8c6556c8d3a7de67/eval/final/components.py#L371-L385

Should I split it into 2 separate parameters?

Reply:
yeah. I think we should separate it. We should not make the assumption that 2 similar but different process always need the same params.